src/Donations/LegacyListeners/DispatchGiveRecurringAddSubscriptionPaymentAndRecordPayment.php

```diff
@@ -22,8 +26,12 @@ public function __invoke(Donation $donation)
         $payment = new Give_Payment($donation->id);
         $parent = new Give_Payment(give()->subscriptions->getInitialDonationId($donation->subscriptionId));
 
+        if (! $donation->type->isRenewal()) {
+            return;
+        }
```

Comment:
Let's put this right at the top. No point in grabbing the subscription and such if we're just going to bail a moment later.

Reply:
I moved this to the service provider.